### ISSUE TITLE
Plugins that reference Post via ForeignKey all erroring when trying to add/edit them

### ISSUE THREAD

**Issue by gmenih341** [nephila/djangocms-blog#369]
I made 3 plugins a few months back, that reference the `Post` model of this app using the `ForeignKey` field. I'm not sure what changed, as no changes were made to the code base, but for some reason all of the plugins that have these fields in them are now returning this error:

```
__str__ returned non-string (type NoneType)

Error during template rendering

In template /home/****/.env/venv/lib/python3.5/site-packages/django/contrib/admin/templates/admin/includes/fieldset.html, error at line 19

__str__ returned non-string (type NoneType)
9	            {% for field in line %}
10	                <div{% if not line.fields|length_is:'1' %} class="field-box{% if field.field.name %} field-{{ field.field.name }}{% endif %}{% if not field.is_readonly and field.errors %} errors{% endif %}{% if field.field.is_hidden %} hidden{% endif %}"{% elif field.is_checkbox %} class="checkbox-row"{% endif %}>
11	                    {% if not line.fields|length_is:'1' and not field.is_readonly %}{{ field.errors }}{% endif %}
12	                    {% if field.is_checkbox %}
13	                        {{ field.field }}{{ field.label_tag }}
14	                    {% else %}
15	                        {{ field.label_tag }}
16	                        {% if field.is_readonly %}
17	                            <p>{{ field.contents }}</p>
18	                        {% else %}
19	
                                  
      {{ field.field }}
      

      
20	                        {% endif %}
21	                    {% endif %}
22	                    {% if field.field.help_text %}
23	                        <p class="help">{{ field.field.help_text|safe }}</p>
24	                    {% endif %}
25	                </div>
26	            {% endfor %}
27	        </div>
28	    {% endfor %}
29	</fieldset>
```
The model here is very simple:

```
class FailingModel(CMSPlugin):
    post = models.ForeignKey(Post, related_name="failing_model", on_delete=models.CASCADE)
```

Any help would be greatly appreciated.

**Comment by gmenih341** [comment]
Update: Looks like it had something to do with one specific blog post. After removing the last added blog post, all the plugins started working again.

**Comment by yakky** [comment]
@gmenih341 maybe something related to the language the blog post was published in? Which version of djangocms blog?

**Comment by gmenih341** [comment]
@yakky Yep, that's exactly it. Just did some further debugging. The version is `0.8.12`, and the problem was that if there were no titles in all available languages, the plugins would fail. After adding only the title (but not publishing the posts), the plugins started working again.

How could I solve this in a better way, so I don't have to add titles in every language?

**Comment by yakky** [comment]
@gmenih341 we need a smarter __str__ I guess. Let me check and I'll be back to you later today

**Comment by yakky** [comment]
@gmenih341 could you test the linked pr if it solves your issue?

Thanks!

**Comment by gmenih341** [comment]
@yakky Used a backed up database, since I wasn't able to create a post without a title again, and it works now - shows the title of a different translation for the post that doesn't exist in current translation.

Thanks for solving this!

**Comment by yakky** [comment]
I need to wait to fixed unrelated test failures before merging it, but thanks for posting this and testing the fix!
I will release a 0.8.13 as soon as the PR will land

**Comment by yakky** [comment]
Fixed in #371 released with 0.8.13
Thanks @gmenih341 for reporting!
